components/datetime/src/time_zone.rs

```diff
@@ -605,12 +605,10 @@ impl FormatTimeZone for SpecificLocationFormat {
             return Ok(Err(FormatTimeZoneError::Fallback));
         };
 
-        if zone_variant == ZoneVariant::daylight() {
-            &locations.pattern_daylight
-        } else if zone_variant == ZoneVariant::standard() {
-            &locations.pattern_standard
-        } else {
-            &locations.pattern_generic
+        match zone_variant {
+            ZoneVariant::Standard => &locations.pattern_standard,
+            ZoneVariant::Daylight => &locations.pattern_daylight,
+            _ => unreachable!(),
```

Comment:
Nit: Annotate here and elsewhere that this is unreachable due to the tilde dependency